src/leetCodeSolutionProvider.ts

```diff
@@ -1,49 +1,28 @@
 // Copyright (c) jdneo. All rights reserved.
 // Licensed under the MIT license.
 
-import * as hljs from "highlight.js";
-import * as MarkdownIt from "markdown-it";
-import * as path from "path";
-import * as vscode from "vscode";
 import { Disposable, ExtensionContext, ViewColumn, WebviewPanel, window } from "vscode";
-import { leetCodeChannel } from "./leetCodeChannel";
 import { IProblem } from "./shared";
+import { MarkdownEngine } from "./webview/MarkdownEngine";
 
 class LeetCodeSolutionProvider implements Disposable {
 
     private context: ExtensionContext;
     private panel: WebviewPanel | undefined;
-    private markdown: MarkdownIt;
-    private markdownPath: string; // path of vscode built-in markdown extension
+    private markdown: MarkdownEngine;
     private solution: Solution;
 
     public initialize(context: ExtensionContext): void {
         this.context = context;
-        this.markdown = new MarkdownIt({
-            linkify: true,
-            typographer: true,
-            highlight: this.codeHighlighter.bind(this),
-        });
-        this.markdownPath = path.join(vscode.env.appRoot, "extensions", "markdown-language-features");
-
-        // Override code_block rule for highlighting in solution language
-        // tslint:disable-next-line:typedef
-        this.markdown.renderer.rules["code_block"] = (tokens, idx, options, _, self) => {
-            const highlight: string = options.highlight(tokens[idx].content, undefined);
-            return [
-                `<pre><code ${self.renderAttrs(tokens[idx])} >`,
-                highlight || this.markdown.utils.escapeHtml(tokens[idx].content),
-                "</code></pre>",
-            ].join("\n");
-        };
+        this.markdown = new MarkdownEngine();
```

Comment:
nit: `markdown` -> `markdownEngine` or `mdEngine` or `engine`.

Reply:
Changed to mdEngine